### PR TITLE
Issue #6207: Add xpath regression test for AvoidNestedBlocks

### PR DESCRIPTION
**Pull request by mincong-h**
This PR adds XPath regression test for suppression filter on [AvoidNestedBlocks](https://checkstyle.org/config_blocks.html#AvoidNestedBlocks) check, see #6207.

Also, it fixes a complication error in related code example in documentation.

```
AvoidNestedBlocks $ cat Demo1.java
public class Demo1 {
  public static void main(String[] args) {
    int a = 0;
    {
      int a = 1;
    }
    System.out.println("a=" + a);
  }
}
```
```
AvoidNestedBlocks $ javac Demo1.java
Demo1.java:5: error: variable a is already defined in method main(String[])
      int a = 1;
          ^
1 error
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
  * {
  *   int whichIsWhich = 0;
  *   {
- *     int whichIsWhich = 2;
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Please revert. Without introduction a new variable, the output is obvious.

Comment (mincong-h) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: I'm ok to revert. But before doing that, I'd like to know what is the added value of this check if there is a compilation error?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: @pbludov ,
I do not understand request. I see change as correct. Example is not real code that is why it is weird. It is ok to change same variable name in inner block, `int ` is a mistake.
Please review one more time.

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: @mincong-h you are right. Java does not allow to "override" variables in the inner scope, as c++ does. But this means that the example is bad. There is no need to guess anything.
But this is not related to this issue.

**Pull request merged**